lib/swaps/Swaps.ts

```diff
@@ -233,10 +290,89 @@ class Swaps extends EventEmitter {
       createTime: Date.now(),
     };
 
+    // add the deal. Going forward we can "record" errors related to this deal.
     this.addDeal(deal);
 
+    // verify LND setup. Make sure we are connected to BTC and LTC and that
+    // the peer is also connected to these networks.
+    const err = this.verifyLNDSetup(deal, peer);
+    if (err) {
+      this.setDealState(deal, SwapDealState.Error, err);
+      this.sendErrorToPeer(peer, deal.r_hash, deal.stateReason,  requestPacket.header.id);
```

Comment:
I see the back to back `setDealState` then `sendErrorToPeer` many times and it seems like we will always do it this way. Maybe the two can be combined in a single function that handles everything - simplest would just be to set the deal state when we send the error to peer. 

Also very minor nit but there are two spaces instead of one before `requestPacket` everywhere this line appears, fix at your leisure.

Reply:
Fixed the 2 spaces. Thanks.

setDealState may be used without sending Error to Peer.
Don't think these should be combined.